lib/src/wiggle_abi/shielding.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::Error;
+use crate::session::Session;
+use crate::wiggle_abi::{fastly_shielding, types};
+
+impl fastly_shielding::FastlyShielding for Session {
+    fn shield_info(
+        &mut self,
+        memory: &mut wiggle::GuestMemory<'_>,
+        name: wiggle::GuestPtr<str>,
+        out_buffer: wiggle::GuestPtr<u8>,
+        _out_buffer_max_len: u32,
+    ) -> Result<u32, Error> {
+        // Validate the input name and then return the unsupported error.
+        let name_bytes = memory.to_vec(name.as_bytes())?;
+        let name = String::from_utf8(name_bytes).map_err(|_| Error::InvalidArgument)?;
+
+        let mut shield_config = name.split('-');
+
+        let first = shield_config.next().expect("first is required");
+        let second = shield_config.next().expect("second is required");
+        let _third = shield_config.next().expect("third is required");
```

Comment:
It also seems that three pieces aren't always present in Shield names. So this is wrong.